src/ewatercycle/_forcings/caravan.py

```diff
@@ -0,0 +1,240 @@
+
+        if 'lat' not in variables and 'lon' not in variables:
+            shape_obj = shapereader.Reader(shape)
+            centre = next(shape_obj.geometries()).centroid
+            ds_basin_time.coords.update({'lat': centre.y,
+                                         'lon': centre.x
+                                         })
+        if variables == ():
+            variables = (ds_basin_time.data_vars.keys())
+
+        # only return the properties which are also in property vars
+        properties = tuple(set(variables).intersection(set(PROPERTY_VARS)))
+        variables = tuple(
+            set(variables).difference(set(PROPERTY_VARS)).intersection(
+              set(RENAME_ERA5.keys())
+            )) # only take the vars also in Rename dict
+
+        for prop in properties:
+            ds_basin_time.coords.update({prop: ds_basin_time[prop].to_numpy()})
+
+        ds_basin_time = ds_basin_time.rename(RENAME_ERA5)
+        variables = ([RENAME_ERA5[var] for var in variables])
+
+        for temp in ['tas','tasmin','tasmax']:
+            ds_basin_time[temp].attrs.update({'height':'2m'})
+
+        for var in variables:
+            ds_basin_time[var].to_netcdf(Path(directory) / f'{basin_id}_{start_time}_{end_time}_{var}.nc')
+
+        forcing = cls(
+            directory=Path(directory),
+            start_time=start_time,
+            end_time=end_time,
+            shape=Path(shape),
+            filenames={
+                var: f'{basin_id}_{start_time}_{end_time}_{var}.nc' for var in variables
+            }
+        )
+        forcing.save()
+        return forcing
+
+
+class LumpedCaravanForcing(Caravan, LumpedUserForcing):  # type: ignore[misc]
+    ...
+
+
+def get_shapefiles(directory: Path, basin_id: str):
+    """Retrieves shapefiles from data 4TU."""
+    zip_path = directory / 'shapefiles.zip'
+    output_path = directory / 'shapefiles'
+
+    if not zip_path.is_file():
+        timeout = 30
+        try:
+            result = requests.get(SHAPEFILE_URL, timeout=timeout)
+        except requests.exceptions.Timeout:
+            msg = (
+            f"Issue connecting to {SHAPEFILE_URL} after {timeout}s"
+            )
+            raise RuntimeError(msg)
+
+        with zip_path.open('wb') as fin:
+            fin.write(result.content)
+
+    combined_shapefile_path = output_path / "combined.shp"
+    if not combined_shapefile_path.is_file():
+        with zipfile.ZipFile(zip_path) as myzip:
+            myzip.extractall(path=directory)
+
+    shape_path = output_path/ f'{basin_id}.shp'
+
+    shape_obj = shapereader.Reader(combined_shapefile_path)
+    list_records = []
+    for record in shape_obj.records():
+        list_records.append(record.attributes['gauge_id'])
+
+    df = pd.DataFrame(data=list_records, index=range(len(list_records)), columns=['basin_id'])
+    basin_index = df[df['basin_id'] == basin_id].index.array[0]
+
+    with fiona.open(
```

Comment:
Perhaps you can refactor this to a separate function to make the main function more readable.